plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import path, { resolve as resolvePath } from 'path';
+import { UrlReader } from '@backstage/backend-common';
+import { InputError } from '@backstage/errors';
+import { ScmIntegrations } from '@backstage/integration';
+import { fetchContents } from './helpers';
+import { createTemplateAction } from '../../createTemplateAction';
+import globby from 'globby';
+import nunjucks from 'nunjucks';
+import fs from 'fs-extra';
+
+export function createFetchTemplateAction(options: {
+  reader: UrlReader;
+  integrations: ScmIntegrations;
+}) {
+  const { reader, integrations } = options;
+
+  return createTemplateAction<{
+    url: string;
+    targetPath?: string;
+    values: any;
+  }>({
+    id: 'fetch:template',
+    description:
+      "Downloads a skeleton and will template variables into the skeleton and places the result in the workspace, or optionally in a subdirectory specified by the 'targetPath' input option.",
+    schema: {
+      input: {
+        type: 'object',
+        required: ['url'],
+        properties: {
+          url: {
+            title: 'Fetch URL',
+            description:
+              'Relative path or absolute URL pointing to the directory tree to fetch',
+            type: 'string',
+          },
+          targetPath: {
+            title: 'Target Path',
+            description:
+              'Target path within the working directory to download the contents to. Defaults to the working directory root.',
+            type: 'string',
+          },
+          values: {
+            title: 'Template Values',
+            description: 'Values to pass on to the templating engine',
+            type: 'object',
+          },
+          // TODO(mtlewis/orkohunter): add copyWithoutRender support
+          //
+          // TODO(mtlewis/orkohunter): do we need to replicate the template extensions support
```

Comment:
Nope - I'd say if they're using that then they should use the cookeicutter action.

Maybe we can also rename `copyWithoutRender` too.

Reply:
But if you mean't should we provide the ability to add some filters, then yes, but maybe later. Let's keep the scope simpler for now and iterate.

Reply:
Makes sense, have removed this todo for now!